src/main.rs

```diff
@@ -323,8 +390,16 @@ async fn main() -> anyhow::Result<()> {
         interval.tick().await;
 
         let rate = {
-            let rates_lock = rates_mutex.lock().unwrap();
-            match compute_median(&*rates_lock) {
+            // For each source, we compute the median of their history:
+            let rate_medians = rate_histories
+                .iter()
+                .map(|rates_mutex| {
+                    let rates_lock = rates_mutex.lock().unwrap();
+                    compute_median(&*rates_lock)
+                })
```

Comment:
I thought we didn't agree to do that, but instead that we kept using old data? :thinking:

Reply:
I see there is a comment added in the test document that I missed when I gave a thumbs up (I was meaning to react to Caro's comment).

I don't really agree with this approach. It introduces unclean behaviour in case of failure, especially in light of the volatility of the price.

Reply:
I have changed it to ignore a source, if it has not been able to read since the last update.